server/support_packet.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"path/filepath"
+	"time"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/plugin"
+)
+
+func (p *Plugin) GenerateSupportData(ctx *plugin.Context) ([]*model.FileData, error) {
+	remoteStorageDir := filepath.Join(pluginDataDir, PluginName, tsdbDirName)
+	dump, err := p.createDump(context.TODO(), ctx.RequestId, time.Now().Add(-1*time.Duration(*p.configuration.SupportPacketMetricsDays)*24*time.Hour), time.Now(), remoteStorageDir)
```

Comment:
Isn't this synchronous? Leading to the same problem again?

Reply:
Yes, but making this asynchronous is requires implementation on server side. We are already taking a conservative approach here for getting the dump for a single day by default.

Alternatively, 
a) we can always prepare a ready dump (it can be generated in let's say every 2 hrs?)
b) use the latest dump from filestore

Reply:
Either of those alternatives sound good to me. But consistency wise, it would look weird to have one async and the other sync.

Reply:
@agnivade changed the behaviour to pick latest successful job (dump). Maybe we can create a dump instead returning an error, but it can take some time. Maybe we can consider implementing an async support packet generation in server side, then we wouldn't need to return an error if there is no ready dump.